version: '3.3'
services:
  postgres:
    image: "postgres"
    environment:
      - POSTGRES_USER=${.ENV}
      - POSTGRES_PASSWORD=${.ENV}
      - POSTGRES_DB=${.ENV}
    ports:
      - "5432:5432"
    volumes:
      - ./postgres-data:/var/lib/postgresql/data
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U postgres"]
      interval: 30s
      timeout: 30s
      retries: 3  
    networks:
      - mynetwork

  api:
    build: 
      context: .
      dockerfile: dockerfile
    ports:
      - "8080:8080"
    depends_on:
      - postgres       
    environment:
      - STRING_CONNECTDB=${.ENV}
    networks:
      - mynetwork

  pgadmin-compose:
    image: dpage/pgadmin4
    environment:
      PGADMIN_DEFAULT_EMAIL: ${.ENV}
      PGADMIN_DEFAULT_PASSWORD: ${.ENV}
    ports:
      - "54321:80"
    depends_on:
      - postgres
    networks:
      - mynetwork
  
networks:
  mynetwork:
    driver: bridge